homeassistant/components/lutron/switch.py

```diff
@@ -50,3 +61,47 @@ def update(self):
         """Call when forcing a refresh of the device."""
         if self._prev_state is None:
             self._prev_state = self._lutron_device.level > 0
+
+
+class LutronLed(LutronDevice, SwitchDevice):
+    """Representation of a Lutron Keypad LED."""
+
+    def __init__(self, area_name, keypad_name, scene_device, led_device, controller):
+        """Initialize the switch."""
+        self._keypad_name = keypad_name
+        self._scene_name = scene_device.name
+        super().__init__(area_name, led_device, controller)
+
+    def turn_on(self, **kwargs):
+        """Turn the LED on."""
+        self._lutron_device.state = 1
+
+    def turn_off(self, **kwargs):
+        """Turn the LED off."""
+        self._lutron_device.state = 0
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        attr = {
+            "keypad": self._keypad_name,
+            "scene": self._scene_name,
+            "led": self._lutron_device.name,
+        }
+        return attr
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        return self._lutron_device.last_state
+
+    @property
+    def name(self):
+        """Return the name of the LED."""
+        return f"{self._area_name} {self._keypad_name}: {self._scene_name} LED"
+
+    def update(self):
+        """Call when forcing a refresh of the device."""
+        if self._lutron_device.last_state is not None:
+            return self._lutron_device.last_state
+        return self._lutron_device.state
```

Comment:
Nothing is checking the return value of `update`.

`self._lutron_device.last_state` has no side effects it just returns the last state.
```suggestion
            return
        self._lutron_device.state
```

Reply:
Just a side note here:
Doing I/O in properties is not friendly for readers of the code. Actions like I/O should be explicit and should be done in functions. I recommend refactoring the library to not use properties for I/O.